Zend/zend_gc.h

```diff
@@ -46,6 +46,20 @@ ZEND_API bool gc_enabled(void);
 ZEND_API bool gc_protect(bool protect);
 ZEND_API bool gc_protected(void);
 
+/* Arrays are non-collectable by default and will be marked as collectable only
+ * when they can form a cycle. This is only the case then they contain objects
+ * or references. */
+static zend_always_inline bool gc_ht_check_collectable(HashTable *ht) {
+	return (GC_FLAGS(ht) & (GC_PERSISTENT|GC_NOT_COLLECTABLE)) == GC_NOT_COLLECTABLE;
+}
```

Comment:
The name of the function is confusing. It doesn't tell what kind of check we do (positive or negative). I would propose to change it into ``gc_ht_is_not_collectable()`` or may be better into ``gc_ht_is_collectable()`` with corresponding condition changes.